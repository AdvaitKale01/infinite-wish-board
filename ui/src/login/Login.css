.login-page {
    position: absolute;
    height: 100%;
    width: 100%;
    background-color: #87cacf;
    color: #2d348d;
    font-family: 'Arial Rounded MT Bold', sans-serif;
    padding: 2.5em;
    box-sizing: border-box;
}

h1 {
    font-size: 3.5em;
}

.header-title {
    align-self: center;
    font-size: 5em;
}

img.header-make-a-wish-logo {
    max-height: 3em;
}

.login-container{
    display: flex;
    justify-content: space-evenly;
}

.registration-form{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.registration-form > input{
    margin-bottom: 10px;
}

.login-form{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.login-form > input{
    margin-bottom: 10px;
}