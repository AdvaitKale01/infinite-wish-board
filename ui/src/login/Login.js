import React, { Component, Fragment } from 'react'
import './Login.css'
import axios from 'axios'
import { withRouter } from 'react-router-dom'

class Login extends Component {
  constructor(props) {
    super(props)

    this.state = {
      username: '',
      password: ''
    }
  }

  createUser = async e => {
    e.preventDefault()

    const { username, password } = this.state

    try {
      await axios.post('http://localhost:3002/auth/register', {
        username,
        password
      })

      this.setState({
        username: '',
        password: ''
      })
    } catch (e) {
      console.log(`response is: ${e}`)
    }
  }

  loginUser = async e => {
    e.preventDefault()

    const { username, password } = this.state

    try {
      await axios
        .create({ withCredentials: true })
        .post('http://localhost:3002/auth/login', {
          username,
          password
        })

      this.props.history.push('/landing')
    } catch (e) {
      console.log(`response is: ${e}`)
    }
  }

  handleFormFieldChange = (key, { target: { value } }) => {
    this.setState({ [key]: value })
  }

  render() {
    return (
      <Fragment>
        <form onSubmit={this.createUser} data-register-form>
          <h2>Register a user</h2>
          Username:{' '}
          <input
            type="text"
            data-register-username
            onChange={event => this.handleFormFieldChange('username', event)}
          />
          Password:{' '}
          <input
            type="password"
            data-register-password
            onChange={event => this.handleFormFieldChange('password', event)}
          />
          <input type="submit" value="Submit!" />
        </form>

        <form onSubmit={this.loginUser} data-login-form>
          <h2>Login</h2>
          Username:{' '}
          <input
            type="text"
            data-login-username
            onChange={event => this.handleFormFieldChange('username', event)}
          />
          Password:{' '}
          <input
            type="password"
            data-login-password
            onChange={event => this.handleFormFieldChange('password', event)}
          />
          <input type="submit" value="Submit!" />
        </form>
      </Fragment>
    )
  }
}

export { Login }
export default withRouter(Login)
