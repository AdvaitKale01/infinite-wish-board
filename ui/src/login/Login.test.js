import React from 'react'
import { Login } from './Login'
import { shallow } from 'enzyme'

describe('When render login page', () => {
  const wrapper = shallow(<Login />)

  it('Should render a Header component', () => {
    expect(wrapper.find('Header').length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render ValidatedSignInForm', () => {
    expect(wrapper.find('ValidatedSignInForm').length).toEqual(1)
  })

  it('Should render ValidatedSignUpForm when clicked on "Create an Account" button', () => {
    wrapper.instance().toggleForm()
    expect(wrapper.find('ValidatedSignUpForm').length).toEqual(1)
    expect(wrapper.state().showSignIn).toBe(false)
  })
})
