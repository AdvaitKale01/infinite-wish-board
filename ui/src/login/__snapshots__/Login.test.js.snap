// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When render login page Should render a Header component 1`] = `ShallowWrapper {}`;
